engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/schemaregistry/confluent/formatter/ConfluentAvroMessageReader.scala

```diff
@@ -1,94 +1,64 @@
 package pl.touk.nussknacker.engine.avro.schemaregistry.confluent.formatter
 
+import io.circe.Json
 import io.confluent.kafka.schemaregistry.avro.AvroSchema
 import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient
 import io.confluent.kafka.serializers.AbstractKafkaAvroSerializer
 import org.apache.avro.Schema.Type
 import org.apache.avro.io.DecoderFactory
 import org.apache.avro.util.Utf8
-import org.apache.avro.{AvroRuntimeException, Schema}
-import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.avro.Schema
 import org.apache.kafka.common.errors.SerializationException
+import pl.touk.nussknacker.engine.avro.AvroUtils
 import pl.touk.nussknacker.engine.avro.schema.DatumReaderWriterMixin
 import pl.touk.nussknacker.engine.avro.schemaregistry.confluent.ConfluentUtils
 
-import java.io.IOException
-import java.nio.charset.StandardCharsets
+import scala.reflect.ClassTag
 
 /**
-  * This class is mainly copy-paste of Confluent's AvroMessageReader but with better constructor handling
-  * both passing schemaRegistryClient and keySeparator.
-  *
   * @param schemaRegistryClient schema registry client
   * @param topic topic
-  * @param keySeparator key separator
   */
-private[confluent] class ConfluentAvroMessageReader(schemaRegistryClient: SchemaRegistryClient,
-                                                    topic: String,
-                                                    useStringForKey: Boolean,
-                                                    keySeparator: String)
+private[confluent] class ConfluentAvroMessageReader(schemaRegistryClient: SchemaRegistryClient, topic: String)
```

Comment:
`topic` is not used, after removing it you can in factory pass just `reader` and not create reader for each record

Reply:
True, fixed.